Example/Tests/Doubles/BitmovinPlayerTestDouble.swift

```diff
@@ -9,157 +9,168 @@
         guard let onStallStarted = fakeListener?.onStallStarted else {
             return
         }
-        onStallStarted(StallStartedEvent())
+        onStallStarted(StallStartedEvent(), player)
     }
 
     func fakeStallEndedEvent() {
         guard let onStallEnded = fakeListener?.onStallEnded else {
             return
         }
-        onStallEnded(StallEndedEvent())
+        onStallEnded(StallEndedEvent(), player)
+    }
+
+    func fakePlayerErrorEvent() {
+        guard let onPlayerError = fakeListener?.onPlayerError else {
+            return
+        }
+        onPlayerError(PlayerErrorEvent(code: PlayerError.Code.networkGeneral, message: "Test player error", data: nil), player)
     }
 
-    func fakeErrorEvent() {
-        guard let onError = fakeListener?.onError else {
+    func fakeSourceErrorEvent() {
+        guard let onSourceError = fakeListener?.onSourceError else {
             return
         }
-        onError(ErrorEvent(code: 1000, message: "Test Error"))
+        onSourceError(SourceErrorEvent(code: SourceError.Code.general, message: "Test source error", data: nil), player)
     }
 
     func fakeAdStartedEvent(position: String? = "pre") {
         guard let onAdStarted = fakeListener?.onAdStarted else {
             return
         }
         onAdStarted(AdStartedEvent(clickThroughUrl: URL(string: "www.google.com")!,
-                                   clientType: .IMA,
+                                   clientType: .ima,
                                    indexInQueue: 1,
                                    duration: 2,
                                    timeOffset: 3,
                                    skipOffset: 4,
                                    position: position,
-                                   ad: BitmovinPlayerTestAd()))
+                                   ad: BitmovinPlayerTestAd()), player)
     }
 
     func fakeTimeChangedEvent() {
         guard let onTimeChanged = fakeListener?.onTimeChanged else {
             return
         }
-        onTimeChanged(TimeChangedEvent(currentTime: 1))
+        onTimeChanged(TimeChangedEvent(currentTime: 1), player)
     }
 
     func fakeSourceUnloadedEvent() {
         guard let onSourceUnloaded = fakeListener?.onSourceUnloaded else {
             return
         }
-        onSourceUnloaded(SourceUnloadedEvent())
+        onSourceUnloaded(SourceUnloadedEvent(source: fakeSource), player)
     }
 
     func fakeAdSkippedEvent() {
         guard let onAdSkipped = fakeListener?.onAdSkipped else {
             return
         }
-        onAdSkipped(AdSkippedEvent(ad: BitmovinPlayerTestAd()))
+        onAdSkipped(AdSkippedEvent(ad: BitmovinPlayerTestAd() as Ad), player)
     }
 
     func fakeAdFinishedEvent() {
         guard let onAdFinished = fakeListener?.onAdFinished else {
             return
         }
-        onAdFinished(AdFinishedEvent(ad: BitmovinPlayerTestAd()))
+        onAdFinished(AdFinishedEvent(ad: BitmovinPlayerTestAd() as Ad), player)
```

Comment:
This cast to `Ad` should also not be needed.

Reply:
Done